test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -4204,6 +4204,31 @@ public async Task Restore_WithHttpsSourceIndexJsonReturningHttpResources_LogsNU1
             result.LockFile.LogMessages.Should().ContainSingle(m => m.Code == NuGetLogCode.NU1302 && m.Message.Contains("http://"));
         }
 
+        [Theory]
+        [InlineData("../")]
+        [InlineData("../my package")]
+        [InlineData("Some.Package/../newtonsoft.json")]
+        public async Task RestoreCommand_WithInvalidPackageId_ShouldFailAsync(string dependency)
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+
+            var project1spec = ProjectTestHelpers.GetPackageSpec("Project1",
+                pathContext.SolutionRoot,
+                framework: "net5.0",
+                dependencyName: dependency);
+
+            var logger = new TestLogger();
+            var command = new RestoreCommand(ProjectTestHelpers.CreateRestoreRequest(pathContext, logger, project1spec));
+
+            // Act
+            RestoreResult result = await command.ExecuteAsync();
+
+            // Assert
+            result.Success.Should().BeFalse();
+            result.LogMessages.Should().Contain(m => m.Code == NuGetLogCode.NU1017 & m.Message == string.Format(Strings.Error_invalid_packageid, dependency));
```

Comment:
Do you mean to do the comparison with the logical AND operator as opposed to the conditional AND operator? If we're using `&`, the second part will always be evaluated even if the first part is false, which doesn't seem necessary in the context of checking if the log messages contain the code we're looking for, which if it doesn't, we already have false as our answer. Please help me understand if I'm missing something! :-)

```suggestion
            result.LogMessages.Should().Contain(m => m.Code == NuGetLogCode.NU1017 && m.Message == string.Format(Strings.Error_invalid_packageid, dependency));
```

Reply:
You're right. we can use the short-circuit AND (&&) here. That said, I don't think it makes a big difference in this case. If the first expression is false, short-circuiting does save some computation. With the logical AND (&), both expressions are always evaluated, which isn't ideal for performance. But since both expressions are very simple, the impact is negligible.

Reply:
Let's use the && (Logical AND) for when possible please.